lib/http-bridge/src/Cardano/Wallet/HttpBridge/Transaction.hs

```diff
@@ -88,6 +92,9 @@ newTransactionLayer = TransactionLayer
     , estimateMaxNumberOfInputs =
         estimateMaxNumberOfInputsBase @t estimateMaxNumberOfInputsParams
 
+    , validateSelection = \(CoinSelection _ outs _) -> do
+        when (any (\ (TxOut _ c) -> c == Coin 0) outs)
+            $ Left ErrInvalidTxOutAmount
```

Comment:
There's now a logic duplication regarding tx validation. The same thing is done in `mkStdTx` and that's something we don't want to be duplicated for there's a risk of diverging as we maintain the code.
My suggestion:

- Change `ErrInvalidTx` to:

```hs
-- | Possible signing error
data ErrMkStdTx
    = ErrKeyNotFoundForAddress Address
    -- ^ We tried to sign a transaction with inputs that are unknown to us?
    | ErrInvalidTx ErrValidateSelection
    -- ^ when transaction with 0 amount is tried (not valid in Byron)
    deriving (Eq, Show)
```

- Call `validateSelection` from within `mkStdTx` and wrap the error accordingly.

Reply:
yes, it is very sensible to do it